### PR TITLE
test: expand full-project coverage

### LINKED ISSUE #1257: Add test coverage for utility functions and shared components
Description
We should improve confidence in the codebase by adding test coverage for shared utility functions and reusable UI components.

Scope
- Add tests for ~~all~~ most important and critical utility/helper functions
- Add tests for shared components
- Cover common edge cases and expected failure states
- Prioritize components and utilities used across multiple flows

## Acceptance criteria
- increase test coverage to >85% across all metrics (lines, statements, functions, branches)
- increase accepted coverage threshold to 80% (see https://github.com/joinmarket-webui/jam/blob/devel/vitest.config.ts#L19-L23)

### PR DESCRIPTION
**Pull request by parrth20**
Expand test coverage across the project

Adds and deepens unit tests across the app to raise coverage toward the project thresholds. Focuses on previously under-tested components, hooks, and context providers - covering loading/error/empty states, conditional branches, and user-interaction flows.

Coverage (combined(unit + storybook)): Statements 90.6% · Lines 90.7% · Functions 89.1% · Branches 82.7%

Also raises the coverage thresholds in the Vitest config and stabilizes a few timing-sensitive tests for reliable CI runs.

closes #1257

### CODE REVIEW

File: src/components/earn/EarnForm.test.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+    render(<EarnForm isWaitingMakerStart={false} offerMinsizeMax={1} onSubmit={onSubmit} />)
+
+    await user.click(screen.getByRole('button', { name: 'earn.button_start' }))
+
+    expect(await screen.findByText('earn.feedback_invalid_min_amount_insufficient_funds')).toBeInTheDocument()
+    expect(onSubmit).not.toHaveBeenCalled()
+  })
+
+  it('renders waiting and debug state', () => {
```
Comment (theborakompanioni) [inline src/components/earn/EarnForm.test.tsx]: Getting a lot of these:
```
An update to TestComponent inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
  /* fire events that update state */
});
/* assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
```

File: src/components/earn/report/EarnReportContent.tsx [hunk 2]
```diff
@@ -119,6 +119,7 @@ export const EarnReportContent = ({ className, enabled }: EarnReportContentProps
     [t],
   )
 
+  // eslint-disable-next-line react-hooks/incompatible-library -- TanStack Table's useReactTable() is not React Compiler compatible
```
Comment (theborakompanioni) [inline src/components/earn/report/EarnReportContent.tsx]: Please do not silence this warning! We _want_ to know that the library is incompatible and verify that it is fixed once a compatible version is released.

Comment (parrth20) [reply src/components/earn/report/EarnReportContent.tsx]: I've removed the eslint-disable comments from all four useReactTable()

File: vitest.setup.ts [hunk 3]
```diff
@@ -2,6 +2,35 @@ import * as matchers from '@testing-library/jest-dom/matchers'
 import { cleanup } from '@testing-library/react'
 import { expect, afterEach } from 'vitest'
 
+// Node 24+ ships an experimental built-in `localStorage` that is disabled (and
+// throws "invalid localStorage location") unless `--localstorage-file` is
+// provided. When that global is active it shadows jsdom's implementation, which
+// breaks persisted zustand stores in the unit tests. Install a simple in-memory
+// Storage so `localStorage` always works in this environment.
+const createMemoryStorage = (): Storage => {
```
Comment (theborakompanioni) [inline vitest.setup.ts]: It seems that if you `export NODE_OPTIONS="--no-webstorage"` mocking localstorage can be skipped. Can you verify?

File: src/components/ui/jam/TablePagination.test.tsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  })
+  Object.defineProperty(HTMLElement.prototype, 'releasePointerCapture', {
+    configurable: true,
+    value: vi.fn(),
+  })
+  Object.defineProperty(HTMLElement.prototype, 'scrollIntoView', {
+    configurable: true,
+    value: vi.fn(),
+  })
+})
```
[the remark this answers is not in the document]
Comment (theborakompanioni) [reply src/components/ui/jam/TablePagination.test.tsx]: ping @parrth20

File: vitest.setup.ts (hunk 3 above)
Comment (parrth20) [reply vitest.setup.ts]: I’ll verify the NODE_OPTIONS approach and remove the localStorage mock if it works.

Comment (parrth20) [reply vitest.setup.ts]: Fixed both points(one of copilot). Prototype methods are now restored after the pagination suite, and I verified --no-webstorage under Node 26.3.0, so the custom localStorage mock is now removed.

Comment (theborakompanioni) [reply vitest.setup.ts]: So nice! Great work! :muscle:

**Review by theborakompanioni — APPROVED** [review]

**Review by volkkov — COMMENTED** [review]
Approach is solid. Minor nit: this would benefit from a brief comment explaining why we skip validation for internal callers.
